support-frontend/assets/helpers/forms/paymentIntegrations/oneOffContributions.ts

```diff
@@ -178,33 +206,41 @@ const processStripePaymentIntentRequest = (
 				}
 
 				// No 3DS auth required
-				return createIntentResponse;
+				return _createIntentResponse;
 			},
 		),
 	);
 
 // Object is expected to have structure:
 // { type: "error", error: PayPalApiError }, or
-// { type: "success", data: PaypalPaymentSuccess }
+// { type: "success", data: CreatePayPalPaymentSuccess }
 function createPayPalPaymentResponseFromObject(
-	res: Record<string, any>,
+	res: Record<string, unknown>,
 ): CreatePayPalPaymentResponse {
-	if (res.data && res.data.approvalUrl) {
-		return {
-			type: 'success',
-			data: {
-				approvalUrl: res.data.approvalUrl,
-			},
-		};
+	if ('data' in res) {
```

Comment:
Why `if ('data' in res)` rather than `if (res.data)`?

Reply:
Here they can be used interchangably, happy to go with one over the other. With `data' in res` we'd get a boolean but with `res.data` we'd also get a truthy or falsy value

Reply:
The only place we'd _need_ to use the "in" method is for `if ('billingAccountId' in existingPaymentMethod)`. As the type of the `existingPaymentMethod` parameter is made up of a union type. Might be better to use this alternative method only where necessary so have used dot notation wherever possible.